### PR TITLE
ci: use GH app for helm workflows

### LINKED ISSUE #14258: Helm release CI broken?
**Describe the bug**
Seems like last two helm charts version (6.13.0, 6.14.0) weren't released properly? https://github.com/grafana/loki/actions/runs/11030863951/job/30636372416

**To Reproduce**
N/A

**Expected behavior**
Helm chart 6.14.0 should be available

**Environment:**
N/A

**Screenshots, Promtail config, or terminal output**
If applicable, add any output to help explain your problem.

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

Change helm workflows to use GH App for auth.

**Which issue(s) this PR fixes**:
Fixes #14258 

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: .github/workflows/helm-ci.yml [hunk 1]
```diff
@@ -27,6 +27,14 @@ jobs:
       - name: Lint Yaml
         run: make helm-lint
 
+      - id: "get_github_app_token"
+        name: "get github app token"
```
Comment (chaudum) [inline .github/workflows/helm-ci.yml:31]: IMO we should use title case for step names:

```suggestion
        name: "Get Github app token"
```

**Review by chaudum — APPROVED** [review]
Otherwise LGTM

**Comment by Starefossen** [comment]
CI is failing with `Error: Input required and not supplied: app-id` ref. https://github.com/grafana/loki/actions/runs/11066924233/job/30749176890?pr=14281

**Comment by trevorwhitney** [comment]
hmm, `app-id` is definitely being provided and it's passing [some builds](https://github.com/grafana/loki/actions/workflows/helm-ci.yml) so something for use to keep an eye on I guess?

**Pull request merged**